### PR TITLE
Experimenting with moving/coalescing desktop docs.

### PR DESCRIPTION
**Pull request by sfshaza2**
It's a beginning that attempts to separate platform specific docs and sort them into directories under Development.

Staged version:  https://sz-flutter.firebaseapp.com/

Most changes are concentrated in **Development -> Platform integrations**. 

Redirects are in place, so if you type, for example, https://sz-flutter.firebaseapp.com/desktop, it does the right thing.

This is a short term solution. Mindy is working on UXR to help figure out an overall solution.

cc @jmagman @InMatrix @MiraMarshall @timsneath @RedBrogdon

### CODE REVIEW

File: src/deployment/windows.md [hunk 1]
```diff
@@ -10,117 +10,152 @@ Flutter app to the [Microsoft Store][microsoftstore].
 ## Preliminaries
 
 Before beginning the process of releasing 
-a Flutter Windows Desktop app, it is necessary
-to confirm that it satisfies [Microsoft Store Policies][storepolicies].
+a Flutter Windows desktop app, first confirm
+that it satisfies [Microsoft Store Policies][storepolicies].
 
-Joining the [Microsoft Partner Network][microsoftpartner] is also required.
+Also, join the
```
Comment (timsneath) [inline src/deployment/windows.md:16]: This sounds like an endorsement, versus the old text which just called it out as a prerequisite.

**Comment by InMatrix** [comment]
Cc: @mthoover

**Review by jmagman — COMMENTED** [review]
It's really hard to tell what was actually changed vs line length changes.  Could you rearrange this to make the first commit just line length changes, then make all the actual edits on top so we can tell which is which?

File: src/development/ios-14.md [hunk 2]
```diff
@@ -1,101 +0,0 @@
...
-
-If your iOS 14 app uses system fonts such as San Francisco (used by default
-by Cupertino widgets), text will be incorrectly rendered in a condensed letter
-spacing due to changes in iOS's font loading mechanism. This affects debug and
-production apps on iOS 14.
-
-To ensure correct font rendering, you should build your production apps with
-Flutter 1.22 beta.
-
-## Debugging Flutter
```
[the remark this answers is not in the document]
Comment (sfshaza2) [reply src/development/ios-14.md:42]: I've moved it, but do we still need it? Is anyone still developing on an earlier release than Flutter 1.21?

File: src/deployment/windows.md (hunk 1 above)
Comment (sfshaza2) [reply src/deployment/windows.md:27]: Changed to "You must join the..."

File: src/development/ios-14.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (timsneath) [reply src/development/ios-14.md:42]: No, I don't think so. Our dashboard suggests that <1% of Flutter developers are running anything before 1.22. At most, a couple of hundred users. Let's remove this.

**Review by timsneath — APPROVED** [review]
LGTM! There's a little more disentangling that could be done, but this is already a major improvement and a foundation for more targeted fixes. Thanks so much, Shams!

File: src/development/ios-14.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmagman) [reply src/development/ios-14.md:42]: I mean, the "Debugging Flutter" section about the local network permissions is relevant on any version of Flutter if you run it on iOS 14 or later.

File: src/development/platform-integration/ios/ios-debugging.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
+title: iOS debugging
+description: iOS-specific debugging techniques for Flutter apps
+---
+
+{{site.alert.note}}
```
Comment (jmagman) [inline src/development/platform-integration/ios/ios-debugging.md:6]: This note can be removed.

Comment (sfshaza2) [reply src/development/platform-integration/ios/ios-debugging.md:6]: ok

**Review by domesticmouse — APPROVED** [review]

File: src/development/platform-integration/ios/apple-watch.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+[bitcode]: {{site.repo.flutter}}/wiki/Creating-an-iOS-Bitcode-enabled-app
+
+## Step 2: Add an Apple Watch target
+
+In the menu, select **File > New > Target**.
+Once the dialog opens, select **watchOS**
+at the top and click **Watch App for iOS App**.
+Click **Next**, enter a product name, and select **Enter**.
+
+![Adding an Apple Watch target]({{site.url}}/assets/images/docs/AppleWatchTarget.png){:width="70%"}
```
[the remark this answers is not in the document]
Comment (sfshaza2) [reply src/development/platform-integration/ios/apple-watch.md:24]: Actually, this is fine on my local version. It just means that I didn't run "DISABLE_TESTS=1 make build" again. I confirm that the file is in place and named correctly.

File: firebase.json [hunk 5]
```diff
@@ -151,9 +163,7 @@
       { "source": "/go/android-embedding-move", "destination": "https://docs.google.com/document/d/1nQujwZfEe3QOHTyZn160eImpoJOYioADP09DtumcLcc/edit?ts=5d8041db#", "type": 301 },
       { "source": "/go/android-migration-summary", "destination": "https://docs.google.com/document/d/1wKspwf6LQu6uo32uQ9NcukfiKhLL4ButV9cwpjeb8QI", "type": 301 },
       { "source": "/go/android-ndk-version", "destination": "https://docs.google.com/document/d/1sljfqHpI5m4uxsL7B89yhYluO97Mz1YKyA0VgzzXyZ4/edit?usp=sharing&resourcekey=0-Ts9Rk8jbJor7WBMMiehKWg", "type": 301 },
-      { "source": "/go/android-plugin-migration.", "destination": "/development/packages-and-plugins/plugin-api-migration", "type": 301 },
-      { "source": "/go/android-plugin-migration", "destination": "/development/packages-and-plugins/plugin-api-migration", "type": 301 },
-      { "source": "/go/android-project-migration", "destination": "https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects", "type": 301 },
```
Comment (xster) [inline firebase.json:156]: This link was completely removed here. I don't think we want to do that. See b/241819690.

**Pull request merged**